src/main/java/com/labsynch/labseer/service/StandardizationServiceImpl.java

```diff
@@ -0,0 +1,524 @@
+				stndznCompound.persist();
+			}
+			// Compute your percentage.
+			percent = (float)Math.floor(p * 100f / totalCount);
+			if(percent != previousPercent)
+			{
+			    // Output if different from the last time.
+			    logger.info("populating standardization dry run table " + percent + "% complete");
+			}
+			// Update the percentage.
+			previousPercent = percent;
+			p++;
+		}
+		logger.info("total number of non matching; structure, display or as drawn display changes: " + nonMatchingCmpds);
+		return (nonMatchingCmpds);
+	}
+
+	@Transactional
+	private String getParentAlias(Parent parent) {
+		StringBuilder aliasSB = new StringBuilder();
+		Set<ParentAlias> parentAliases = parent.getParentAliases();
+		boolean firstAlias = true;
+		for (ParentAlias parentAlias : parentAliases){
+			if (firstAlias){
+				aliasSB.append(parentAlias.getAliasName());
+				firstAlias = false;
+			} else {
+				aliasSB.append(";").append(parentAlias.getAliasName());
+			}
+		}
+		return aliasSB.toString();
+	}
+
+
+	@Override
+	public int dupeCheckStandardizationStructures() throws CmpdRegMolFormatException{
+		List<Long> qcIds = StandardizationDryRunCompound.findAllIds().getResultList();
+		int totalCount = qcIds.size();
+		logger.debug("number of compounds found in dry run table: " + totalCount);
+		int totalNewDuplicateCount = 0;
+		int totalExistingDuplicateCount = 0;
+		if (qcIds.size() > 0){
+			int[] hits;	
+			StandardizationDryRunCompound qcCompound;
+			String newDuplicateCorpNames = "";
+			String oldDuplicateCorpNames = "";
+			int newDupeCount = 0;
+			int oldDuplicateCount = 0;
+			
+			float percent = 0;
+			int p = 1;
+		    float previousPercent = percent;
+			previousPercent = percent;
+			for (Long qcId : qcIds){
+				boolean firstNewDuplicateHit = true;
+				boolean firstOldDuplicateHit = true;
+				qcCompound = StandardizationDryRunCompound.findStandardizationDryRunCompound(qcId);
+				logger.debug("query compound: " + qcCompound.getCorpName());
+				if(qcCompound.getNewMolWeight() == 0) {
+					logger.debug("mol has a weight of 0 - skipping");			
+				} else {
+					hits = chemStructureService.searchMolStructures(qcCompound.getMolStructure(), StructureType.DRY_RUN, SearchType.DUPLICATE_TAUTOMER);
+					newDupeCount = hits.length;
+					qcCompound.setChangedStructure(true);
+					for (int hit:hits){
+						List<StandardizationDryRunCompound> searchResults = StandardizationDryRunCompound.findStandardizationDryRunCompoundsByCdId(hit).getResultList();
+						for (StandardizationDryRunCompound searchResult : searchResults){
+							if (searchResult.getCorpName().equalsIgnoreCase(qcCompound.getCorpName())){
+								qcCompound.setChangedStructure(false);
+								newDupeCount = newDupeCount-1;
+							} else {
+								if (StringUtils.equals(searchResult.getStereoCategory(), qcCompound.getStereoCategory())
+										&& StringUtils.equalsIgnoreCase(searchResult.getStereoComment(), qcCompound.getStereoComment())){
```

Comment:
Does this logic handle cases when one or both are `null`? I recall in other code (probably in BulkLoadServiceImpl) we implemented some utilities or shortcuts for this "is the parent a dupe" logic.

Reply:
Yes, null is handled by the StringUtils functions